ui/admin/app/components/model-providers/ModelProviderForm.tsx

```diff
@@ -0,0 +1,167 @@
+
+    const form = useForm<ModelProviderFormValues>({
+        resolver: zodResolver(formSchema),
+        mode: "onChange",
+        defaultValues: {
+            requiredConfigParams: getInitialRequiredParams(
+                requiredParameters,
+                parameters
+            ),
+            additionalConfirmParams: getInitialAdditionalParams(
+                requiredParameters,
+                parameters
+            ),
+        },
+    });
+
+    const requiredConfigParamFields = useFieldArray({
+        control: form.control,
+        name: "requiredConfigParams",
+    });
+
+    const { execute: onSubmit, isLoading } = useAsync(
+        async (data: ModelProviderFormValues) => {
+            const allConfigParams: Record<string, string> = {};
+            [data.requiredConfigParams, data.additionalConfirmParams].forEach(
+                (configParams) => {
+                    for (const param of configParams) {
+                        const paramValue =
+                            "value" in param ? param.value : param.description;
+                        if (paramValue && param.name) {
+                            allConfigParams[param.name] = paramValue;
+                        }
+                    }
+                }
+            );
+
+            await configureModelProvider(modelProviderId, allConfigParams);
+            onSuccess(allConfigParams);
+        }
+    );
+
+    return (
+        <div className="flex flex-col gap-4">
+            <TypographyH4 className="font-semibold text-md">
+                Required Configuration
+            </TypographyH4>
+            <Form {...form}>
+                <form
+                    onSubmit={form.handleSubmit(onSubmit)}
+                    className="flex flex-col gap-8"
+                >
+                    {requiredConfigParamFields.fields.map((field, i) => (
+                        <ControlledInput
+                            key={field.id}
+                            label={field.name}
+                            control={form.control}
+                            name={`requiredConfigParams.${i}.value`}
+                            classNames={{
+                                wrapper: "flex-auto bg-background",
+                            }}
+                        />
+                    ))}
+                </form>
+            </Form>
+
+            <Separator className="my-4" />
+
+            <TypographyH4 className="font-semibold text-md">
+                Custom Configuration (Optional)
+            </TypographyH4>
+            <NameDescriptionForm
+                defaultValues={form.watch("additionalConfirmParams")}
+                onChange={(values) =>
+                    form.setValue("additionalConfirmParams", values)
+                }
+            />
+
+            <div className="flex justify-end">
+                <Button
+                    onClick={form.handleSubmit(onSubmit)}
```

Comment:
instead of duplicating submission logic, you can use native HTML APIs to remotely trigger a form

```ts
const FORM_ID = "model-provider-form"

<form id={FORM_ID}>...</form>

// notice that button is outside of the form
<Button type="submit" form={FORM_ID} /> // clicking this button will now trigger submit on the above form
```

Reply:
Awesome, thank you!